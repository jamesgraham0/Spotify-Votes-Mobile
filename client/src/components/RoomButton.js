import { StyleSheet, Text, TouchableOpacity } from 'react-native';
import React from 'react';
import * as Haptics from 'expo-haptics';

const RoomButton = ({ onPress, disabled, buttonText, buttonStyle, textStyle }) => {
    return (
        <TouchableOpacity
            onPress={() => {
                Haptics.impactAsync(Haptics.ImpactFeedbackStyle.Heavy);
                onPress();
            }}
            style={[stylesButton.button, buttonStyle]}
            disabled={disabled}
        >
            <Text style={[stylesButton.buttonText, textStyle]}>{buttonText}</Text>
        </TouchableOpacity>
    );
};

const stylesButton = StyleSheet.create({
    button: {
        borderWidth: 2,
        width: 300,
        height: 80,
        borderRadius: 50,
        shadowOpacity: 1,
        shadowOffset: { width: 0, height: 0 },
        shadowRadius: 5,
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#191414',
    },
    buttonText: {
        position: 'absolute',
        color: '#fff',
        fontSize: 24,
    },
});

export default RoomButton;