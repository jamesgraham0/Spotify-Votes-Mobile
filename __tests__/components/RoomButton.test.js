import React from 'react';
import renderer from 'react-test-renderer';
import RoomButton from '../../client/src/components/RoomButton';

describe('RoomButton', () => {
    it('should render the RoomButton component', () => {
        const onPress = jest.fn();
        const disabled = false;
        const buttonText = "Click me";
        const buttonStyle = { backgroundColor: "blue" };
        const textStyle = { color: "white" };

        const tree = renderer.create(
            <RoomButton
                onPress={onPress}
                disabled={disabled}
                buttonText={buttonText}
                buttonStyle={buttonStyle}
                textStyle={textStyle}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});