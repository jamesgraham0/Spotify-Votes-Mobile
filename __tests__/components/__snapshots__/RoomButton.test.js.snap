// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomButton should render the RoomButton component 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "blue",
      "borderRadius": 50,
      "borderWidth": 2,
      "display": "flex",
      "height": 80,
      "justifyContent": "center",
      "opacity": 1,
      "shadowOffset": {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 5,
      "width": 300,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#fff",
          "fontSize": 24,
          "position": "absolute",
        },
        {
          "color": "white",
        },
      ]
    }
  >
    Click me
  </Text>
</View>
`;
